bundles/org.openhab.binding.zwavejs/src/main/java/org/openhab/binding/zwavejs/internal/conversion/BaseMetadata.java

```diff
@@ -0,0 +1,595 @@
+        if (splitted.length < 2) {
+            return 1.0;
+        }
+
+        try {
+            return Double.parseDouble(splitted[0]);
+        } catch (NumberFormatException e) {
+            return 1.0;
+        }
+    }
+
+    protected boolean isAdvanced(int commandClassId, String propertyName) {
+        return COMMAND_CLASSES_ADVANCED.contains(commandClassId);
+    }
+
+    /*
+     * Normalizes the label based on the provided parameters.
+     *
+     * @param label the label
+     * 
+     * @param endpoint the endpoint
+     * 
+     * @param propertyName the property name
+     *
+     * @return the normalized label
+     */
+    private String normalizeLabel(@Nullable String label, int endpoint, String propertyName) {
+        String output = "";
+        if (label == null || label.isBlank()) {
+            return propertyName;
+        }
+        output = label.replaceAll("\s\\[.*\\]", "");
+        output = capitalize(output);
+        if (endpoint > 0) {
+            output = String.format("EP%s %s", endpoint, output);
+        }
+        return output;
+    }
+
+    /*
+     * Capitalizes the input string by splitting camelCase words.
+     *
+     * @param input the input string
+     *
+     * @return the capitalized string
+     */
+    private String capitalize(@Nullable String input) {
+        if (input == null || input.isBlank()) {
+            return DEFAULT_LABEL;
+        }
+
+        return Objects
+                .requireNonNullElse(
+                        Arrays.stream(StringUtils.splitByCharacterType(input)).filter(f -> !f.isBlank())
+                                .map(word -> StringUtils.capitalize(word)).collect(Collectors.joining(" ")),
+                        DEFAULT_LABEL)
+                .replace(" - ", "-").replace("( ", "(").replace(" )", ")");
+    }
+
+    private String normalizeString(@Nullable Object input) {
+        if (input instanceof Number numberInput) {
+            return "-" + numberInput.toString();
+        } else if (input instanceof String strInput) {
+            return "-" + strInput.trim().toLowerCase().replaceAll(" ", "-").replaceAll("[^a-zA-Z0-9\\-]", "");
+        }
+        return "";
+    }
+
+    private String generateId(String commandClassName, int endpoint, @Nullable String propertyName,
+            @Nullable Object propertyKey) {
+        String id = normalizeString(commandClassName).replaceFirst("-", "");
+        String[] splitted;
+        if (propertyName != null && !propertyName.contains("unknown")) {
+            propertyName = CHANNEL_ID_PROPERTY_NAME_REPLACEMENTS.getOrDefault(propertyName, propertyName);
+            splitted = StringUtils.splitByCharacterType(propertyName);
+            List<String> result = Arrays.asList(splitted).stream().filter(s -> s.matches("^[a-zA-Z0-9]+$")).toList();
+            if (!result.isEmpty()) {
+                id += normalizeString(String.join("-", result));
+            }
+        }
+        if (propertyKey != null) {
```

Comment:
As endPoint and propertyKey are integers, I think this can lead to non unique channel ids between endpoint 0 with propertyKey 1 (say) and endpoint 1 with propertyKey null.

Reply:
Actually in retrospect I think this is Ok..

- `current-color1` = main channel of endpoint 1
- `current-color-1` = sub channel of endpoint 0
-

Reply:
Yeah, offcourse I knew that. Totally thought about this. 100% by design 😬